### PR TITLE
Add `tracer_provider` kwarg to `AzureMonitorTraceExporter`

### PR DESCRIPTION
**Pull request by avramdj**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by avramdj** [comment]
https://github.com/Azure/azure-sdk-for-python/issues/36362

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_exporter.py [hunk 1]
```diff
@@ -71,6 +71,10 @@
 class AzureMonitorTraceExporter(BaseExporter, SpanExporter):
     """Azure Monitor Trace exporter for OpenTelemetry."""
 
+    def __init__(self, **kwargs: Any):
+        self._tracer_provider = kwargs.pop("tracer_provider", None)
```
Comment (lzchen) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_exporter.py:75]: Let's use `get` instead so we don't alter the kwargs.

**Comment by lzchen** [comment]
@avramdj 

Thanks for the contribution! Please add a [CHANGELOG](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/monitor/azure-monitor-opentelemetry-exporter/CHANGELOG.md) entry and add a test for this to show that a custom `TracerProvider` can be used.

**Comment by annatisch** [comment]
Hi @avramdj  - just wanted to check in as to whether you wanted to keep working on this PR?
Otherwise @lzchen do you want to finish up this PR?

**Comment by avramdj** [comment]
@microsoft-github-policy-service agree

**Review by lzchen — APPROVED** [review]

**Pull request merged**